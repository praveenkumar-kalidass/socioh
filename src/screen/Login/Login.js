import React from 'react';

import { CONSTANT } from '../../constant';
import { LoginView, Title, TitleView } from './Login.style';

const Login = () => {
  return (
    <LoginView>
      <TitleView>
        <Title>{CONSTANT.APP_NAME}</Title>
      </TitleView>
    </LoginView>
  );
};

export default Login;
