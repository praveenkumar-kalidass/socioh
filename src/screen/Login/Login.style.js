import styled from 'styled-components/native';
import { Text, Icon, Button } from 'native-base';

import { COLOR, FONT, SPACING } from '../../constant';

export const LoginContainer = styled.View`
  background-color: ${COLOR.PRIMARY};
  flex: 1;
  justify-content: space-around;
  padding-horizontal: ${SPACING.SPACE_20};
  padding-vertical: ${SPACING.SPACE_20};
`;

export const Title = styled(Text)`
  color: ${COLOR.WHITE};
  font-family: Roboto;
  font-size: ${FONT.SIZE_60};
  font-weight: bold;
  text-transform: uppercase;
`;

export const TitleContainer = styled.View`
  align-items: center;
`;

export const Description = styled(Text)`
  color: ${COLOR.WHITE};
  font-family: Roboto;
  font-size: ${FONT.SIZE_24};
  text-align: center;
`;

export const IconContainer = styled.View`
  margin-horizontal: ${SPACING.SPACE_40};
`;

export const LargeIcon = styled(Icon)`
  color: ${COLOR.WHITE};
  font-size: ${SPACING.SPACE_200};
`;

export const SmallIcon = styled(Icon)`
  color: ${COLOR.WHITE};
  font-size: ${SPACING.SPACE_100};
  transform: scaleX(-1);
`;

export const ButtonContainer = styled.View`
  padding-horizontal: ${SPACING.SPACE_20};
`;

export const SignInContainer = styled.View`
  flex-direction: row;
  justify-content: center;
`;

export const SignInButton = styled(Button)`
  background-color: ${COLOR.SECONDARY};
  flex: 1;
  justify-content: center;
`;

export const SignInButtonText = styled(Text)`
  color: ${COLOR.BLACK};
  text-align: center;
`;

export const SignUpContainer = styled.View`
  align-items: center;
  margin-top: ${SPACING.SPACE_20};
`;

export const SignUpLink = styled(Button)``;

export const SignUpDescription = styled(Text)`
  color: ${COLOR.WHITE};
  font-family: Roboto;
  font-size: ${FONT.SIZE_16};
`;

export const SignUpText = styled(Text)`
  color: ${COLOR.WHITE};
  text-decoration: underline;
  text-decoration-color: ${COLOR.WHITE};
`;
