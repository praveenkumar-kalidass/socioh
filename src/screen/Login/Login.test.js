import React from 'react';
import { render, act, fireEvent } from '@testing-library/react-native';

import Login from './index';

describe('Login', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<Login />);

    expect(container).toMatchSnapshot();
  });

  it('should navigate on press of signup link', () => {
    const { getByTestId } = render(<Login navigation={mockNavigation} />);

    act(() => fireEvent(getByTestId('sign_up'), 'onPress'));

    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith('SIGNUP');
  });
});
