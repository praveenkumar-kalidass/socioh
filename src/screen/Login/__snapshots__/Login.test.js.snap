// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "#1abfbf",
        "paddingVertical": 40,
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginTop": 40,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 80,
              "textTransform": "uppercase",
            },
          ]
        }
      >
        socioh
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
